src/NServiceBus.Core/Persistence/PersistenceExtentions.cs

```diff
@@ -85,14 +86,22 @@ public PersistenceExtentions(Type definitionType, SettingsHolder settings, Type
                 SelectedStorages = new List<Type>(),
             };
 
+            
             if (storageType != null)
             {
+                var definition = definitionType.Construct<PersistenceDefinition>();
+                if (!definition.HasSupportFor(storageType))
+                {
+                    throw new Exception(string.Format("Persistence '{0}' does not support storage type {1}.", definitionType.Name, storageType.Name));
```

Comment:
I think we should give the user some direction on how to fix this exception, maybe guide them to a doco page that shows them what persistence support what, thoughts?